### PR TITLE
ToCode Solution: תרגול ביטויים רגולריים

### PR DESCRIPTION
**Pull request by omerblecher**

### CODE REVIEW

File: 18_regexp_lab/01.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+        word = line.strip(' \t\n\r')
+        if not word.startswith('#'):
+            keyvalue = re.split(r'[=]', line)
+            defDict[keyvalue[0].strip(' \t\n\r')] = keyvalue[1].strip(' \t\n\r')
+
+#Print the value that mact
+if key in defDict.keys():
+    print defDict[key]
+else:
+    print "%s is not in the definition file" % (key)
```
Comment (szabgab) [inline 18_regexp_lab/01.py]: Good direction but if the propertise file has an empty line I get

```
Traceback (most recent call last):
  File "01.py", line 15, in <module>
    defDict[keyvalue[0].strip(' \t\n\r')] = keyvalue[1].strip(' \t\n\r')
IndexError: list index out of range
```

Also you could replace `keyvalue = re.split(r'[=]', line)` by
`key, value = keyvalue = re.split(r'[=]', line)`` and that would make the code more readable.

Comment (omerblecher) [inline 18_regexp_lab/01.py]: Fix

File: 18_regexp_lab/01.py [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
+            defDict[key.strip(' \t\n\r')] = value.strip(' \t\n\r')
 
 #Print the value that mact
-if key in defDict.keys():
-    print defDict[key]
+if userkey in defDict.keys():
+    print defDict[userkey]
 else:
-    print "%s is not in the definition file" % (key)
+    print "%s is not in the definition file" % (userkey)
```
Comment (szabgab) [inline 18_regexp_lab/01.py:21]: 👍

File: 18_regexp_lab/02.py [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
+
+#From camel case
+def fromCamelCase(text):
+    return re.sub(r'[A-Z]',
+        lambda m: '_' + m.group(0).lower()
+        , text)
+
+
+print toCamelCase('no_more_shall_we_part')
+print  fromCamelCase('noMoreShallWePart')
```
Comment (szabgab) [inline 18_regexp_lab/02.py:17]: Nice, though your roCamelCase has not really used the power of regexes. You could have used regular split as well.

Comment (omerblecher) [inline 18_regexp_lab/02.py:17]: Fixed

File: 18_regexp_lab/02.py [hunk 4]
```diff
@@ -2,9 +2,9 @@
...
 def toCamelCase(text):
-    allWords = re.split(r'_', text)
-    first = allWords[0]
-    return first + ''.join([word.title() for word in allWords[1:]])
+    return re.sub(r'_[a-z]',
+        lambda m: m.group(0)[1].upper()
+        , text)
 
 #From camel case
 def fromCamelCase(text):
```
Comment (szabgab) [inline 18_regexp_lab/02.py:10]: 👍

File: 18_regexp_lab/03.py [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
+#CSV column replacements
+
+
+with open(sys.argv[1], "r") as fin:
+    for line in fin:
+        word = line.strip(' \t\n\r')
+        if len(word) > 0 :
+            res = re.search(r'^(.+),(.+),(.+)$', word) #Get all the column values
+            if res is not None:
+                print "%s,%s,%s" % (res.group(2), res.group(1), res.group(3))
```
Comment (szabgab) [inline 18_regexp_lab/03.py]: Using substitution you would not need to assume there are exactly 3 columns.
Actually I see you don't assume 3 columns, you are just putting all the left-overs in the 3d group.
Still there is an assumption of a second comma. Could you eliminate that assumption (to work in the case of 2 columns only) with a little tweak to the current solution?

File: 18_regexp_lab/03.py [hunk 6]
```diff
@@ -7,6 +7,9 @@
...
-            res = re.search(r'^(.+),(.+),(.+)$', word) #Get all the column values
-            if res is not None:
-                print "%s,%s,%s" % (res.group(2), res.group(1), res.group(3))
\ No newline at end of file
+
+            columns = re.findall(r'(.+?)(?:,|$)', word) #Get all the column values
+            temp = columns[0]
+            columns[0] = columns[1]
+            columns[1] = temp #Swap
+            print ','.join(columns)
```
Comment (szabgab) [inline 18_regexp_lab/03.py:15]: That's probably one of the solutions 😄

**Pull request closed without merging**